### ISSUE TITLE
BlockExecution not working properly in 3.6.2

### ISSUE THREAD

**Issue by Kevinnns** [PSAppDeployToolkit/PSAppDeployToolkit#16]
Hi,

When i add -BlockExecution to the Show-InstallationWelcome -CloseApps i get an error (see attached image).

When i run the appdeploy without the -BlockExecution parameter everything goes well.
Used command: Show-InstallationWelcome -CloseApps 'iexplore.exe,iexplorer.exe,firefox.exe,chrome.exe,eclipse.exe' -BlockExecution -CloseAppsCountdown 600

Can anyone tell me what i'm doing wrong? I Can't seem to find anything in the logs.

![psappdeploy](https://cloud.githubusercontent.com/assets/12066544/7272940/36b12c28-e8ef-11e4-8a7a-16b64e00e0ef.jpg)

**Comment by mmashwani** [comment]
Check out the documentation for the -CloseApps parameter. It says not to include the .exe for the name of the process. Therefore, your command should really be:
```
Show-InstallationWelcome -CloseApps 'iexplore,iexplorer,firefox,chrome,eclipse' -BlockExecution -CloseAppsCountdown 600
```

Try that and let me know if that fixes your problem.

**Comment by Kevinnns** [comment]
Hi mmashwani, thanks for the reply.
My first run was without the .exe and the same behavior was presented.

By the way in the documentation in example 5 the .exe is being used with the -CloseApps parameter.

**Comment by mmashwani** [comment]
Good catch, I'll fix that in the docs. This is a strange one. I can't replicate this issue.

From the error, it seems as if the script is not able to create this file: C:\Users\Public\PSAppDeployToolkit\AppDeployToolkit_BlockAppExecutionMessage.vbs

Can you go to that location and see if you can create an empty file there manually?

Also, maybe try deleting the whole folder, if it exists, and they run the script again?

**Comment by Kevinnns** [comment]
Looking closer, it seems that the creation of the C:\Users\Public\PSAppDeployToolkit directory isn't functioning for me. I get 1 file without extension (after deletion and trying again i get no files/folders).

It seems tho that running install (even without -blockexecution) isn't running for me either (i've attached a screenshot). I'm invoking the Deploy-Application.exe as Administrator in a cmd with admin rights.
![psappdeploy1](https://cloud.githubusercontent.com/assets/12066544/7273414/c24afb48-e8f3-11e4-9eb1-7dd3ce4b7f60.jpg)
![psappdeploy2](https://cloud.githubusercontent.com/assets/12066544/7273413/c249ff54-e8f3-11e4-8224-66455bbef3b6.jpg)

**Comment by mmashwani** [comment]
Can you test on another machine and see if you get the same results? Could there be some sort of security software that is blocking the creation of these files/folders or blocking execution?

**Comment by mmashwani** [comment]
Also, please try the latest 3.6.3 beta from here to see if that works for you? There are a few important bugs this beta addresses:
https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/tree/master/Toolkit/AppDeployToolkit

**Comment by Kevinnns** [comment]
I just tried 3.6.3 and i've tried an other VM/Computer and the same thing happens.
I have .Net framework 4.5.2 installed, could it have something to do with that?

And when you delete the C:\Users\Public\PSAppDeployToolkit, does it get recreated on your system?

**Comment by mmashwani** [comment]
I'm on .NET v4, so not sure if it's .NET related. Yes, that folder will get recreated and files will be overwritten every time you run the toolkit. Do you have a DLP security program in your environment? I have worked in environments where this type of security program caused a lot of mysterious issues because it is almost impossible to detect that it is even running on your system (unless you use process monitor).

**Comment by Kevinnns** [comment]
It's a pretty vanilla Windows 7 VM to create (MSI) packages and script installs with PSAppDepoly on.
So no security software or w/e installed. For testing i'll deploy a new one, just to see what happens.

I'm closing in on the problem with the install (without -blockexecution). That's because the .msi -needs- to be in the Files folder. Because of SCCM i usually throw the .msi in the root folder where the .ps1 and deploy-application.exe lives. Looks like i'm going to stop doing that ;-).

So that only leaves the -blockexecution problem for me. The C:\Users\Public\PSAppDeployToolkit folder doesn't get created on those machines. Instead they get a C:\Users\Public\PSAppDeployToolkit file that looks like AppDeployToolkitMain.ps1 (when opened).

So to be totally clear, does C:\Users\Public\PSAppDeployToolkit need to be a folder or a file? In my case with the 3.6.2/3.6.3 it's a file and looking at older versions it was a folder in the past. Looking at the error it still should be a folder, where a AppDeployToolkit_BlockAppExecutionMessage.vbs file is expected.

**Comment by mmashwani** [comment]
It needs to be a folder. And then all of the toolkit files are copied there. They are required for various async tasks. In 3.6.2, look at line 4421 for the folder creation. Look at line 4424 to for the files being copied into that folder. Maybe do some troubleshooting around these lines to see if you can figure out what is going on. I would suggest setting the -ErrorAction parameter to 'Stop' for lines 4420 to 4424 to see if there is a specific error that gets thrown which might shed more light on the problem.

**Comment by Kevinnns** [comment]
Okay, looks like it's "fixed". My "clean" VMs had a C:\Users\Public\PSAppDeployToolkit file instead of folder (dating back from 2014, version 3.1.1). When i delete that file the new version is able to create a folder again and works like i'm used to. I don't now if this was default behavior in that time, but if it was, it would be great to have a clean-up job implemented?

I thought i removed the file while testing, but reverted to the original snapshot, thus having the file again (d0h). Now i'm running a GPO that removes the "C:\Users\Public\PSAppDeployToolkit" file on all computers.

The reason that my other VMs failed (true vanilla) was due to the requirement of the .msi needs to exist in the Files folder (which was new for me, coming from 3.2.0), that put me on the wrong track in the beginning.

I'll close the ticket in a few hours, since i'm interested in if 3.1.1 did make that file instead of a folder.

mmashwani, thank you for your help/feedback!

**Comment by mmashwani** [comment]
Awesome, glad you figured it out. I think that file might have been some bug from an older version of the toolkit or some remnant of the toolkit from before I got involved with the project.

**Comment by mmashwani** [comment]
I just pushed a fix for this to the 3.6.3 beta. The toolkit will now look for this file and delete it before trying to create a folder in this location.
